dynamic_rest/renderers.py

```diff
@@ -0,0 +1,93 @@
+from collections import defaultdict
+from rest_framework import renderers
+
+
+class DynamicJSONRenderer(renderers.JSONRenderer):
+
+  def render(self, data, accepted_media_type=None, renderer_context=None):
+    # if data is a string, just render it out using the default method
+    # this usually indicates an error message
+    if isinstance(data, basestring):
+      return super(DynamicJSONRenderer, self).render(data, accepted_media_type, renderer_context)
+
```

Comment:
Considering the way you do `errors` in batch create, it might actually be helpful to extend (4) in such a way that the viewset can provide a map of additional data to the renderer/response that it outside of the realm of serializers and should not be modified, e.g. `viewset.extra_data = { 'errors' : ......, 'meta': ...... }` or `return Response(serializer.data, extra_data = {'errors' .... 'meta' .... }`